File: oops/WEEK 2/MatrixAdd.java
import java.util.Scanner;

class Matrix {
    void matrix_add() {
        Scanner in = new Scanner(System.in);

        System.out.print("Enter size of m1: ");
        int m1 = in.nextInt();
        System.out.print("Enter size of n1: ");
        int n1 = in.nextInt();

        System.out.print("Enter size of m2: ");
        int m2 = in.nextInt();
        System.out.print("Enter size of n2: ");
        int n2 = in.nextInt();

        if (m1 != m2 || n1 != n2) {
            System.out.println("Matrices cannot be added.");
            return;
        }

        System.out.println("Enter elements for matrix A:");
        int a[][] = new int[m1][n1];
        for (int i = 0; i < m1; i++) {
            for (int j = 0; j < n1; j++) {
                System.out.print("Enter element (" + (i + 1) + "," + (j + 1) + "): ");
                a[i][j] = in.nextInt();
            }
        }

        System.out.println("Enter elements for matrix B:");
        int b[][] = new int[m2][n2];
        for (int i = 0; i < m2; i++) {
            for (int j = 0; j < n2; j++) {
                System.out.print("Enter element (" + (i + 1) + "," + (j + 1) + "): ");
                b[i][j] = in.nextInt();
            }
        }

        int c[][] = new int[m1][n1];
        System.out.println("Matrix addition result:");
        for (int i = 0; i < m1; i++) {
            for (int j = 0; j < n1; j++) {
                c[i][j] = a[i][j] + b[i][j];
                System.out.print(c[i][j] + "\t");
            }
            System.out.println("\t");
        }

        in.close();
        
    }
}

class MatrixAdd {
    public static void main(String[] args) {
        Matrix add = new Matrix();
        add.matrix_add();
    }
}





























/*class Matrix
{
    void matrix_add()
    {
        int a[][] = {{1,2,3},{4,5,6},{7,8,9}};
        int b[][] = {{1,2,3},{4,5,6},{7,8,9}};
        int c[][] = new int[3][3];
        for (int i=0; i<3; i++)
        {
            for (int j=0; j<3; j++)
            {
                c[i][j] = a[i][j]+b[i][j];
                System.out.println(c[i][j]+" ");
            }
            System.out.println(" ");
        }
    }
}
class MatrixAdd
{
    public static void main(String[] args)
    {
        Matrix add = new Matrix();
        add.matrix_add();
    }
}*/

File: Program1/matrixadd.c
#include<stdio.h>
int main()
{
	int r1,c1,r2,c2,i,j,c[10][10];
	printf("enter row and col size of 1st matrix");
	scanf("%d%d",&r1,&c1);
	printf("enter row and col size of 2nd matrix");
	scanf("%d%d",&r2,&c2);
	int a[r1][c1],b[r2][c2];
	if(r1==r2&&c1==c2)
	{
	printf("enter the elements of first matrix");
	
	for(i=0;i<r1;i++)
	{
	for(j=0;j<c1;j++)	
	{
		scanf("%d",&a[i][j]);
	}
	}
		printf("enter the elements of 2nd matrix");
	
	for(i=0;i<r2;i++)
	{
	for(j=0;j<c2;j++)	
	{
		scanf("%d",&b[i][j]);
	}
	}
	

	for(i=0;i<r1;i++)
	{
	for(j=0;j<c1;j++)	
	{
		c[i][j]=a[i][j]-b[i][j];
	}
	
	}
		printf("addition of matrices");
			for(i=0;i<r1;i++)
	{
	for(j=0;j<c1;j++)	
	{
		printf("%d ",c[i][j]);
	}
	printf("\n");
	
	}
}
else
printf("addition will not be performed");
}
